common/map_test.go

```diff
@@ -0,0 +1,188 @@
+package common
+
+import (
+	"slices"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+//nolint:funlen,gocognit, cyclop
+func TestGenericMap(t *testing.T) {
```

Comment:
once you rename GenericMap to StringMap, don't forget to update this test function also.